# Bad data 제거
**Bad data의 앞 몇글자 가져오기**

In [1]:
# final_text.txt의 index로 fucking data 찾기
fuckingData = [96160, 111236, 114232, 114866, 116346, 133016, 176778]
f=open("rsc/training_data/final_text.txt")
lines = f.readlines()

headLines = []

## 데이터 확인
for lineNum in fuckingData:
    headLines.append(lines[lineNum][:20])

print(headLines)

['목차참고 단위환산표 읽는 법1. 길이', '분류 산하설계 응모안 1. 내진 내풍', '목차(1) 용량(2) 응용IBS(In', '요약 보안 암호화하여 파일을 전송한다', '목차전혀 콤팩트하지 않은 CF(Com', '목차(1) 타이어의 치수 표시(2) ', '순백금 세선을 감는 틀에 감은 저항형']


In [2]:
# csv file 불러오기
import pandas as pd
final = pd.read_csv("../Dataset/FinalDataset.csv")

In [3]:
# headline이 포함된 Data의 인덱스 찾기
fucking_data_index = []

for i in range(len(final)):
    for headLine in headLines:
        if headLine in final.loc[i].values[0]:
            fucking_data_index.append(i)
            print("index : {}".format(i))

print(fucking_data_index)

index : 48080
index : 55618
index : 57116
index : 57433
index : 58173
index : 66508
index : 88389
[48080, 55618, 57116, 57433, 58173, 66508, 88389]


# fucking data Info
**final_text.txt**<br>
index = [48080, 55618, 57116, 57433, 58173, 66508, 88389]
### raw data에서 찾기

In [5]:
# PATH
# INPUT PATH
PAPER_PATH = '../Dataset/Datacon/paper.csv'
KoSAT_PATH = '../Dataset/WebCrawling/수능/ICES_sooneung.csv'
NEWS_PATH = '../Dataset/WebCrawling/news/ICES_News.csv'
VOCA_DIC_PATH = ""

# OTUPUT PATH
FINAL_DATA_CSV_PATH = '../Dataset/finalDataset.csv'
FINAL_DATA_TXT_PATH = "rsc/training_data/final_text.txt"
FINAL_DATA_TXT_SETENCE_PATH = "rsc/training_data/final_text_sentence.txt"
DISTRIBUTED_FINAL_DATA_TXT_SETENCE_PATH = 'rsc/training_data/Finalresult_{}.txt'

In [6]:
# 논문 dataset
paper_data = pd.read_csv(PAPER_PATH)
paper_data.drop(['Unnamed: 0'], axis='columns', inplace=True)
# 수능 dataset
koSAT = pd.read_csv(KoSAT_PATH)
# 웹 크롤링 - 뉴스 dataset
news = pd.read_csv(NEWS_PATH)
# 웹 크롤링 - 단어사전 dataset
import os
file_path = [i for i in os.listdir('../Dataset/WebCrawling/dictionary') if i != '.ipynb_checkpoints' and 'xlsx' not in i]
vocab_dic = pd.DataFrame()
for i in file_path:
    df4 = pd.read_csv('../Dataset/WebCrawling/dictionary/'+i[:-4]+'.csv')
    vocab_dic =  pd.concat([vocab_dic, df4])

In [7]:
import re
def pre_processing(csv_file, key_name):
    csv_file[key_name] = csv_file[key_name].astype(str)
    # tag 제거
    csv_file[key_name] = csv_file[key_name].map(lambda x: re.sub('(<([^>]+)>)', '',x.replace('\n','')))
    # tag 제거2
    csv_file[key_name] = csv_file[key_name].map(lambda x: re.sub(r'&(/?)(.+?);','',x))
    # 영어, 숫자, 연산 기호만 남기기
    csv_file[key_name] = csv_file[key_name].map(lambda x: re.sub('[^a-zA-Z0-9ㄱ-ㅣ가-힣+-/*÷=×±∓∘∙∩∪≅∀√%∄∃θπσ≠<>≤≥≡∼≈≢∝≪≫∈∋∉⊂⊃⊆⊇⋈∑∫∏∞x().,%#{}]',' ',x))
    # 다중 띄어쓰기 제거
    csv_file[key_name] = csv_file[key_name].map(lambda x: re.sub(' +',' ', x))
    
    return csv_file

In [8]:
# 각 데이터 전처리 진행
paper_data = pre_processing(paper_data, 'PAPER_FULL_TXT')
koSAT = pre_processing(koSAT, 'text')
news = pre_processing(news, 'content')
vocab_dic = pre_processing(vocab_dic, 'content')
vocab_dic = vocab_dic.reset_index(drop=True)

In [11]:
# 각 Dataset에서 위치 찾기
for headline in headLines:
    for i in range(len(paper_data)):
        if headline in paper_data.loc[i,'PAPER_FULL_TXT']:
            print("paper data, index : {}".format(i))
            break
    for i in range(len(koSAT)):
        if headline in koSAT.loc[i,'text']:
            print("koSAT data, index : {}".format(i))
            break
    for i in range(len(news)):
        if headline in news.loc[i,'content']:
            print("news data, index : {}".format(i))
            break
    for i in range(len(vocab_dic)):
        if headline in vocab_dic.loc[i,'content']:
            print("vocab_dic, index : {}".format(i))
            break

vocab_dic, index : 1705
vocab_dic, index : 9245
vocab_dic, index : 10743
vocab_dic, index : 11060
vocab_dic, index : 11800
vocab_dic, index : 20137
vocab_dic, index : 42020


In [14]:
# 제거 확인
temp = vocab_dic.drop(vocab_dic.index[[1705, 9245, 10743, 11060, 11800, 20137, 42020]])
print(len(temp))
print(len(vocab_dic))

for i in [1705, 9245, 10743, 11060, 11800, 20137, 42020]:
    print(vocab_dic.loc[i].values[1])

54820
54827
목차참고 단위환산표 읽는 법1. 길이2. 면적3. 부피4. 무게5. 온도6. 속도7. 힘8. 압력9. 응력10. 체적유량11. 일, 에너지 및 열량12. 일률참고 단위환산표 읽는 법예를 들어, 1m를 km로 환산할 경우 좌측 끝의 세로축에서 1m를 선택한다. 상단 가로축에서 변환하려는 단위를 선택한다. 2개에서 각각 수평, 수직선을 내려 겹치는 곳의 값을 읽는다. 이 경우 1m는 0.001km이다.1. 길이단위cmmkminftydmilenm척( )리( )1cm10.010.000010.39370.03280.01096×10 65.4×10 60.0332.5×10 51m10010.00139.373.28081.09360.00065.4×10 43.32.5×10 31km1×1051,00010.39373,280.81,093.60.621370.543,3002.51in2.540.02542.54×10 510.08330.02781.6×10 51.4×10 50.08386.5×10 51ft30.480.30483.04×10 41210.33330.000191.65×10 41.0058776×10 41yd91.4380.91449.14×10 436310.00064.94×10 43.01750.00021mile160,9301609.31.6093633605280176010.8695310.80.40981nm185,2001,8521.85272,9136,0762,0251.15116,1114.7161척( )30.3030.3033.03×10 411.930.99420.33140.00021.65×10 410.000081리( )39,272392.70.392727154611288429.50.2440.2121,2961 (척=자)은 손가락을 펴서 엄지와 장지의 길이에 근거한다. 인치(inch)는 1150년경 스코틀랜드의 데이비드 1세에 의해서 엄지손가락 손톱의 뿌리 부분의 폭으로 정의된 바 있으며, 14세기 초 에드워드 2세의 통치 기간에는 1인치(in)는 마르고 둥근 3개의 보리 낱알을 길이 방